File: Lista 4/Python/Ponto2D.py
class Ponto2D:
    def __init__(self, x = 0, y = 0):
        self.__x = x
        self.__y = y

    def __str__(self):
        return f"({self.__x}, {self.__y})"
    
    @property
    def x(self):
        return self.__x
    
    @property
    def y(self):
        return self.__y
    
    @x.setter
    def x(self, x):
        self.__x = x

    @y.setter
    def y(self, y):
        self.__y = y
    
    def isEixoX(self):
        if self.__x == 0:
            return True
        return False
        
    def isEixoY(self):
        if self.__y == 0:
            return True
        return False
    
    def isEixos(self):
        if self.__x == 0 or self.__y == 0:
            return True
        return False
    
    def quadrante(self):
        if self.isEixos():
            return 0
        
        if self.__x > 0:
            if self.__y > 0:
                return 1
            else:
                return 4
        else:
            if self.__y > 0:
                return 2
            else:
                return 3
            
    def distancia(self, Ponto2D):
        from math import sqrt
        distancia = sqrt((Ponto2D.x - self.__x)**2 + (Ponto2D.y - self.__y)**2)
        return distancia


if __name__ == "__main__":
    ponto = Ponto2D()
    print(ponto)
    ponto.y = 3
    print("IsEixoX:", ponto.isEixoX())
    print("IsEixoY:", ponto.isEixoY())
    print("IsEixos:", ponto.isEixos())

    ponto2 = Ponto2D(4, 0)
    print(ponto.distancia(ponto2))

File: Lista 4/Java/Ponto2D.java
import java.lang.Math;

public class Ponto2D {
    private int x;
    private int y;

    public Ponto2D(int x, int y) {
        this.x = x;
        this.y = y;
    }
    
    public void imprimirPonto() {
        System.out.println("(" + x + ", " + y + ")");
    }

    public boolean isEixoX() {
        return x == 0;
    }

    public boolean isEixoY() {
        return y == 0;
    }

    public boolean isEixos() {
        return (x == 0 || y == 0);
    }

    public int quadrante() {
        if (x == 0 || y == 0)
            return 0;
        
        if (x > 0)
            return ((y > 0) ? 1 : 4);
        else
            return ((y > 0) ? 2 : 3);
    }

    // Getters
    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public void setX(int x) {
        this.x = x;
    }

    public void setY(int y) {
        this.y = y;
    }

    public float distancia(Ponto2D ponto) {
        float distancia = (float) Math.sqrt(((ponto.getX() - x) * (ponto.getX() - x)) + ((ponto.getY() - y) * (ponto.getY() - y)));
        return distancia;
    }

    
    public static void main(String[] args){
        Ponto2D ponto1 = new Ponto2D(5, 1);
        Ponto2D ponto2 = new Ponto2D(0, 3);

        ponto1.imprimirPonto();
        ponto1.setX(4);
        ponto1.setY(0);
        ponto1.imprimirPonto();
        System.out.println(ponto1.isEixoX());
        System.out.println(ponto1.isEixoY());
        System.out.println(ponto1.isEixos());

        System.out.println(ponto2.getX() + ", " + ponto2.getY());
        System.out.println(ponto1.distancia(ponto2));
    }
}
